include/zephyr/drivers/sensor.h

```diff
@@ -806,10 +806,19 @@ struct __attribute__((__packed__)) sensor_data_generic_header {
 	/* Shift value for all samples in the frame */
 	int8_t shift;
 
+	int8_t _padding;
```

Comment:
Is there a simple way to turn this off to save memory on platforms that do support unaligned access?

Reply:
I checked with Fabio and currently there's no Kconfig to tell us if we have support for unaligned access

Reply:
yeah not that I'm aware of anyway, afaik normally the compiler decides on whether to generate unaligned code or not based on the target architecture and flags like `-mno-unaligned-access`, the Kconfig option is just to enable trapping of those for architecture that supports both. I don't think that the compiler has anything that you could use, at least not in the preprocessor, but I may be wrong here, or maybe you could use an align attribute in the field.

Reply:
If we're having alignment issues then we shouldn't be packing this struct.

Reply:
Discussed with Fabio offline, the packing doesn't actually change this, we need to have alignment on the end of the struct with the size of `enum sensor_channel`, packing (or not) won't affect this (I checked). If packing and the `_padding` are removed we still run into the issue.

Reply:
I've tweaked the assert to check for `% sizeof(enum sensor_channel) == 0`

Reply:
OK, 2 problems were making this more difficult than expected:
1. `size_t` isn't always the same, I replaced it with `uint32_t` for the `num_channels` field of the header.
2. `sizeof(enum sensor_channel)` isn't always the same (some platforms used 2 bytes others 4). So I made the `_padding` variable based on it.

Reply:
ouch :-\ how bad would it be to memcpy the samples in local buffers instead?

Reply:
is it the samples or the channel array that's the issue? I'm gathering that it's the channel array is unaligned because of the int8_t and packed attribute?

Reply:
Both